esp/smc/SMCLib/TpWrapper.cpp

```diff
@@ -1570,6 +1576,9 @@ void CTpWrapper::getMachineList(const char* MachineType,
         if (!root)
             throw MakeStringExceptionDirect(ECLWATCH_CANNOT_GET_ENV_INFO, MSG_FAILED_GET_ENVIRONMENT_INFO);
 
+        bool hasPropChannelsPerNode = root->hasProp("@channelsPerNode");
```

Comment:
You could move these down to closer to where they are used, or do them inline.

Reply:
It is used inside a while loop. It may not be efficient if it is called many times.

Reply:
Ah, ok, that's fine.